### ISSUE TITLE
Crash on open

### ISSUE THREAD

**Issue by reisub** [fman-community/fman#164]
OS X 10.12.5, fman 0.4.7

More details:

```
Process:               fman [13640]
Path:                  /Applications/fman.app/Contents/MacOS/fman
Identifier:            io.fman.fman
Version:               0.4.7 (0.4.7)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           fman [13640]
User ID:               501

Date/Time:             2017-05-24 17:19:58.236 +0200
OS Version:            Mac OS X 10.12.5 (16F73)

Time Awake Since Boot: 33000 seconds
Time Since Wake:       300 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
*** Terminating app due to uncaught exception 'SUNoFeedURL', reason: 'You must specify the URL of the appcast as the SUFeedURL key in either the Info.plist or the user defaults!'
terminating with uncaught exception of type NSException
abort() called

Application Specific Backtrace 1:
0   CoreFoundation                      0x00007fffc7f952cb __exceptionPreprocess + 171
1   libobjc.A.dylib                     0x00007fffdcda048d objc_exception_throw + 48
2   CoreFoundation                      0x00007fffc8013c3d +[NSException raise:format:] + 205
3   Sparkle                             0x000000011048f1b1 -[SUUpdater feedURL] + 340
4   Sparkle                             0x000000011048d3f5 -[SUUpdater checkIfConfiguredProperly] + 249
5   Sparkle                             0x000000011048e88f -[SUUpdater checkForUpdatesWithDriver:] + 523
6   libdispatch.dylib                   0x0000000104be9524 _dispatch_call_block_and_release + 12
7   libdispatch.dylib                   0x0000000104be08fc _dispatch_client_callout + 8
8   libdispatch.dylib                   0x0000000104bedaac _dispatch_main_queue_callback_4CF + 925
9   CoreFoundation                      0x00007fffc7f4abc9 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
10  CoreFoundation                      0x00007fffc7f0bc0d __CFRunLoopRun + 2205
11  CoreFoundation                      0x00007fffc7f0b114 CFRunLoopRunSpecific + 420
12  HIToolbox                           0x00007fffc746cebc RunCurrentEventLoopInMode + 240
13  HIToolbox                           0x00007fffc746ccf1 ReceiveNextEventCommon + 432
14  HIToolbox                           0x00007fffc746cb26 _BlockUntilNextEventMatchingListInModeWithFilter + 71
15  AppKit                              0x00007fffc5a05a54 _DPSNextEvent + 1120
16  AppKit                              0x00007fffc61817ee -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 2796
17  AppKit                              0x00007fffc59fa3db -[NSApplication run] + 926
18  libqcocoa.dylib                     0x0000000109b098e5 qt_plugin_instance + 110581
19  QtCore                              0x0000000107845b91 _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE + 417
20  QtCore                              0x0000000107849f27 _ZN16QCoreApplication4execEv + 343
21  PyQt5.QtWidgets.so                  0x000000010855d5e0 _ZL23meth_QApplication_exec_P7_objectS0_ + 80
22  Python                              0x000000010566eae9 PyCFunction_Call + 233
23  Python                              0x00000001056fba81 PyEval_EvalFrameEx + 35729
24  Python                              0x00000001056fc4fc PyEval_EvalFrameEx + 38412
25  Python                              0x00000001056fce90 _PyEval_EvalCodeWithName + 2400
26  Python                              0x00000001056fcff1 PyEval_EvalCode + 81
27  fman                                0x0000000103f6f23d fman + 8765
28  fman                                0x0000000103f6f771 fman + 10097
29  fman                                0x0000000103f6e1a4 fman + 4516

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x0000000104dc4d42 __pthread_kill + 10
1   libsystem_pthread.dylib       	0x0000000104f19457 pthread_kill + 90
2   libsystem_c.dylib             	0x0000000104ce7420 abort + 129
3   libc++abi.dylib               	0x00007fffdc26f84a abort_message + 266
4   libc++abi.dylib               	0x00007fffdc294c4f default_terminate_handler() + 267
5   libobjc.A.dylib               	0x00007fffdcda26fe _objc_terminate() + 103
6   libc++abi.dylib               	0x00007fffdc291d69 std::__terminate(void (*)()) + 8
7   libc++abi.dylib               	0x00007fffdc291de3 std::terminate() + 51
8   libdispatch.dylib             	0x0000000104be0910 _dispatch_client_callout + 28
9   libdispatch.dylib             	0x0000000104bedaac _dispatch_main_queue_callback_4CF + 925
10  com.apple.CoreFoundation      	0x00007fffc7f4abc9 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
11  com.apple.CoreFoundation      	0x00007fffc7f0bc0d __CFRunLoopRun + 2205
12  com.apple.CoreFoundation      	0x00007fffc7f0b114 CFRunLoopRunSpecific + 420
13  com.apple.HIToolbox           	0x00007fffc746cebc RunCurrentEventLoopInMode + 240
14  com.apple.HIToolbox           	0x00007fffc746ccf1 ReceiveNextEventCommon + 432
15  com.apple.HIToolbox           	0x00007fffc746cb26 _BlockUntilNextEventMatchingListInModeWithFilter + 71
16  com.apple.AppKit              	0x00007fffc5a05a54 _DPSNextEvent + 1120
17  com.apple.AppKit              	0x00007fffc61817ee -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 2796
18  com.apple.AppKit              	0x00007fffc59fa3db -[NSApplication run] + 926
19  libqcocoa.dylib               	0x0000000109b098e5 0x109ae7000 + 141541
20  QtCore                        	0x0000000107845b91 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 417
21  QtCore                        	0x0000000107849f27 QCoreApplication::exec() + 343
22  PyQt5.QtWidgets.so            	0x000000010855d5e0 meth_QApplication_exec_(_object*, _object*) + 80
23  Python                        	0x000000010566eae9 PyCFunction_Call + 233
24  Python                        	0x00000001056fba81 PyEval_EvalFrameEx + 35729
25  Python                        	0x00000001056fc4fc PyEval_EvalFrameEx + 38412
26  Python                        	0x00000001056fce90 _PyEval_EvalCodeWithName + 2400
27  Python                        	0x00000001056fcff1 PyEval_EvalCode + 81
28  io.fman.fman                  	0x0000000103f6f23d 0x103f6d000 + 8765
29  io.fman.fman                  	0x0000000103f6f771 0x103f6d000 + 10097
30  io.fman.fman                  	0x0000000103f6e1a4 0x103f6d000 + 4516

Thread 1:
0   libsystem_kernel.dylib        	0x0000000104dc544e __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x0000000104f16621 _pthread_wqthread + 1426
2   libsystem_pthread.dylib       	0x0000000104f1607d start_wqthread + 13

Thread 2:
0   libsystem_kernel.dylib        	0x0000000104dc544e __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x0000000104f1648e _pthread_wqthread + 1023
2   libsystem_pthread.dylib       	0x0000000104f1607d start_wqthread + 13

Thread 3:
0   libsystem_kernel.dylib        	0x0000000104dc544e __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x0000000104f16621 _pthread_wqthread + 1426
2   libsystem_pthread.dylib       	0x0000000104f1607d start_wqthread + 13

Thread 4:
0   libsystem_kernel.dylib        	0x0000000104dc544e __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x0000000104f16621 _pthread_wqthread + 1426
2   libsystem_pthread.dylib       	0x0000000104f1607d start_wqthread + 13

Thread 5:
0   libsystem_kernel.dylib        	0x0000000104dc4bf2 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x0000000104f177fa _pthread_cond_wait + 712
2   Python                        	0x000000010574108b PyThread_acquire_lock_timed + 475
3   Python                        	0x0000000105747149 acquire_timed + 137
4   Python                        	0x000000010574729d lock_PyThread_acquire_lock + 61
5   Python                        	0x000000010566eb19 PyCFunction_Call + 281
6   Python                        	0x00000001056fba81 PyEval_EvalFrameEx + 35729
7   Python                        	0x00000001056fce90 _PyEval_EvalCodeWithName + 2400
8   Python                        	0x00000001056fc3ef PyEval_EvalFrameEx + 38143
9   Python                        	0x00000001056fce90 _PyEval_EvalCodeWithName + 2400
10  Python                        	0x00000001056fc3ef PyEval_EvalFrameEx + 38143
11  Python                        	0x00000001056fce90 _PyEval_EvalCodeWithName + 2400
12  Python                        	0x00000001056fcf97 PyEval_EvalCodeEx + 71
13  Python                        	0x000000010564977a function_call + 186
14  Python                        	0x00000001056168d3 PyObject_Call + 99
15  Python                        	0x00000001056f5813 PyEval_EvalFrameEx + 10531
16  Python                        	0x00000001056fce90 _PyEval_EvalCodeWithName + 2400
17  Python                        	0x00000001056fcf97 PyEval_EvalCodeEx + 71
18  Python                        	0x000000010564977a function_call + 186
19  Python                        	0x00000001056168d3 PyObject_Call + 99
20  Python                        	0x00000001056f5813 PyEval_EvalFrameEx + 10531
21  Python                        	0x00000001056fce90 _PyEval_EvalCodeWithName + 2400
22  Python                        	0x00000001056fc3ef PyEval_EvalFrameEx + 38143
23  Python                        	0x00000001056fc4fc PyEval_EvalFrameEx + 38412
24  Python                        	0x00000001056fce90 _PyEval_EvalCodeWithName + 2400
25  Python                        	0x00000001056fcf97 PyEval_EvalCodeEx + 71
26  Python                        	0x000000010564977a function_call + 186
27  Python                        	0x00000001056168d3 PyObject_Call + 99
28  Python                        	0x00000001056319cc method_call + 140
29  Python                        	0x00000001056168d3 PyObject_Call + 99
30  Python                        	0x00000001056f1217 PyEval_CallObjectWithKeywords + 87
31  Python                        	0x0000000105746c47 t_bootstrap + 71
32  libsystem_pthread.dylib       	0x0000000104f1693b _pthread_body + 180
33  libsystem_pthread.dylib       	0x0000000104f16887 _pthread_start + 286
34  libsystem_pthread.dylib       	0x0000000104f1608d thread_start + 13

Thread 6:
0   libsystem_pthread.dylib       	0x0000000104f16070 start_wqthread + 0
1   ???                           	0x000000000000ceb3 0 + 52915

Thread 7:: QFileInfoGatherer
0   libsystem_kernel.dylib        	0x0000000104dc4bf2 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x0000000104f177fa _pthread_cond_wait + 712
2   QtCore                        	0x000000010768d92b 0x107663000 + 174379
3   QtCore                        	0x000000010768d8c2 QWaitCondition::wait(QMutex*, unsigned long) + 162
4   QtWidgets                     	0x0000000108d54e52 0x108b03000 + 2432594
5   QtCore                        	0x000000010768c3aa 0x107663000 + 168874
6   libsystem_pthread.dylib       	0x0000000104f1693b _pthread_body + 180
7   libsystem_pthread.dylib       	0x0000000104f16887 _pthread_start + 286
8   libsystem_pthread.dylib       	0x0000000104f1608d thread_start + 13

Thread 8:: QFileInfoGatherer
0   libsystem_kernel.dylib        	0x0000000104dc4bf2 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x0000000104f177fa _pthread_cond_wait + 712
2   QtCore                        	0x000000010768d92b 0x107663000 + 174379
3   QtCore                        	0x000000010768d8c2 QWaitCondition::wait(QMutex*, unsigned long) + 162
4   QtWidgets                     	0x0000000108d54e52 0x108b03000 + 2432594
5   QtCore                        	0x000000010768c3aa 0x107663000 + 168874
6   libsystem_pthread.dylib       	0x0000000104f1693b _pthread_body + 180
7   libsystem_pthread.dylib       	0x0000000104f16887 _pthread_start + 286
8   libsystem_pthread.dylib       	0x0000000104f1608d thread_start + 13

Thread 9:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	0x0000000104dbd34a mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x0000000104dbc797 mach_msg + 55
2   com.apple.CoreFoundation      	0x00007fffc7f0c434 __CFRunLoopServiceMachPort + 212
3   com.apple.CoreFoundation      	0x00007fffc7f0b8c1 __CFRunLoopRun + 1361
4   com.apple.CoreFoundation      	0x00007fffc7f0b114 CFRunLoopRunSpecific + 420
5   com.apple.AppKit              	0x00007fffc5b52f02 _NSEventThread + 205
6   libsystem_pthread.dylib       	0x0000000104f1693b _pthread_body + 180
7   libsystem_pthread.dylib       	0x0000000104f16887 _pthread_start + 286
8   libsystem_pthread.dylib       	0x0000000104f1608d thread_start + 13

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000006  rcx: 0x00007fff5bc8f648  rdx: 0x0000000000000000
  rdi: 0x0000000000000307  rsi: 0x0000000000000006  rbp: 0x00007fff5bc8f670  rsp: 0x00007fff5bc8f648
   r8: 0x0000000104d10650   r9: 0x0000000104cd2a50  r10: 0x0000000008000000  r11: 0x0000000000000206
  r12: 0x00007fff5bc8f7d0  r13: 0x0000000000000030  r14: 0x0000000104f1e3c0  r15: 0x0000000000000008
  rip: 0x0000000104dc4d42  rfl: 0x0000000000000206  cr2: 0x0000000104f60d50
  
Logical CPU:     0
Error Code:      0x02000148
Trap Number:     133
```

**Comment by reisub** [comment]
This happened only once, I did not reproduce it since.

EDIT: it happens intermittently, twice so far.

**Comment by mherrmann** [comment]
Interesting. Seems other people have this problem as well: https://github.com/openscad/openscad/issues/1588

**Comment by mherrmann** [comment]
Should be fixed in fman 0.4.8, which just went live.
